config/initializers/config_check.rb

```diff
@@ -51,3 +51,30 @@
 config_yml_file = "./config/settings.#{ENV['CKB_NET_MODE']}.yml"
 Config.load_and_set_settings config_yml_file
 
+def check_environments
+
+  # rule 1: CKB_NET_MODE and CKB_NODE_URL must exist
+  if ENV['CKB_NET_MODE'].blank? || ENV['CKB_NODE_URL'].blank?
+    raise "environment: CKB_NET_MODE or CKB_NET_MODE not found, please check '.env' and restart."
+  end
+
+  # rule 2: CKB_NET_MODE and CKB_NODE_URL must match
+  command = %Q{
+  echo '{
+      "id": 1,
+      "jsonrpc": "2.0",
+      "method": "get_blockchain_info",
+      "params": []
+  }' | tr -d '\n' | curl -H 'content-type: application/json' -d @- #{ENV['CKB_NODE_URL']}
+  }
+
+  node_mode_from_json_rpc = JSON.parse(`#{command}`)['result']['chain'] rescue nil
```

Comment:
We shall not spawn shell command to obtain chain mode.
In case there were no these commands in environment.